src/test/coins_tests.cpp

```diff
@@ -133,36 +147,62 @@ BOOST_AUTO_TEST_CASE(coins_cache_simulation_test)
         // Do a random modification.
         {
             uint256 txid = txids[insecure_rand() % txids.size()]; // txid we're going to modify in this iteration.
-            CCoins& coins = result[txid];
-            CCoinsModifier entry = stack.back()->ModifyCoins(txid);
-            BOOST_CHECK(coins == *entry);
-            if (insecure_rand() % 5 == 0 || coins.IsPruned()) {
-                if (coins.IsPruned()) {
-                    added_an_entry = true;
+            Coin& coin = result[COutPoint(txid, 0)];
+            Coin newcoin;
+            const Coin& entry = (insecure_rand() % 500 == 0) ? AccessByTxid(*stack.back(), txid) : stack.back()->AccessCoin(COutPoint(txid, 0));
+            BOOST_CHECK(coin == entry);
+
+            if (insecure_rand() % 5 == 0 || coin.IsPruned()) {
+                newcoin.out.scriptPubKey.assign(insecure_rand() & 0x3F, 0); // Random sizes so we can test memory usage accounting
+                newcoin.out.nValue = insecure_rand();
+                newcoin.nHeight = 1;
+            }
+            if (newcoin.IsPruned()) {
```

Comment:
In commit "Extend coins_tests"

I think it would be clearer to combine this if/else block with the previous one, so the different cases perform steps in a uniform order, and there are fewer conditions that need to be checked. The following seems to do the same work and drops around 10 lines of code:

```
if (insecure_rand() % 5 == 0 || coin.IsPruned()) {
    newcoin.out.nValue = insecure_rand();
    newcoin.nHeight = 1;
    if (insecure_rand() % 16 == 0 && coin.IsPruned()) {
        newcoin.out.scriptPubKey.assign(1 + (insecure_rand() & 0x3F), OP_RETURN);
        BOOST_CHECK(newcoin.out.scriptPubKey.IsUnspendable());
        added_an_unspendable_entry = true;
    } else {
        newcoin.out.scriptPubKey.assign(insecure_rand() & 0x3F, 0); // Random sizes so we can test memory usage accounting
        (coin.IsPruned() ? added_an_entry : updated_an_entry) = true;
        coin = newcoin;
    }
    stack.back()->AddCoin(COutPoint(txid, 0), std::move(newcoin), !coin.IsPruned() || insecure_rand() & 1);
} else {
    removed_an_entry = true;
    coin.Clear();
    stack.back()->SpendCoin(COutPoint(txid, 0));
}
```